### ISSUE TITLE
opencv2/tracking/tracking_legacy.hpp requires private headers

### ISSUE THREAD

**Issue by luiscarox** [opencv/opencv#19260]
##### System information (version)

- OpenCV => 4.5.1
- Operating System / Platform => Ubuntu 18.04
- Compiler => gcc 8.4

##### Detailed description

In order to use `cv::legacy::Tracker` (https://docs.opencv.org/master/db/dfe/classcv_1_1legacy_1_1Tracker.html), I get a compiler error when including `opencv2/tracking/tracking_legacy.hpp`

```
In file included from /home/xxx/opencv-5.4.1/install/include/opencv4/opencv2/tracking/tracking_legacy.hpp:55
...
/home/xxx/opencv-5.4.1/install/include/opencv4/opencv2/tracking/tracking_internals.hpp:18:10: fatal error: opencv2/video/detail/tracking.private.hpp: No such file or directory
 #include "opencv2/video/detail/tracking.private.hpp"
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

##### Steps to reproduce
* Build OpenCV with opencv_contrib, ensure the tracking module is built
* Run the install target and install into a location
* Try to include `tracking_legacy.hpp` from the installed location (and NOT the opencv source folder)

The following code will trigger the compiler error:

```
#include <opencv2/tracking.hpp>
#include <opencv2/tracking/tracking_legacy.hpp>
```

##### Issue submission checklist

 - [x] I report the issue, it's not a question
   
 - [x] I checked the problem with documentation, FAQ, open issues,
       answers.opencv.org, Stack Overflow, etc and have not found solution
   
 - [x] I updated to latest OpenCV version and the issue is still there
   
 - [ ] There is reproducer code and related data files: videos, images, onnx, etc

**Comment by ferdnyc** [comment]
On an unrelated note (well, it's at least tangentially related, as it regards the release process that got us here), I'm curious why the migration of the previous tracking API from `cv::Tracker` to `cv::legacy::Tracker` was released as part of OpenCV 4.5.1. That seems like an API-breaking change (even absent this header snafu) that would warrant a minor version bump, but none of this was part of 4.5.0.

**Comment by alalek** [comment]
"opencv_contrib" modules have no requirements about API/ABI strong compatibility, header names, etc.
